### PR TITLE
docs: data types

### PR DESCRIPTION
**Pull request by v0y4g3r**
Add docs for SQL data types.

### CODE REVIEW

File: docs/reference/data-types.md [hunk 1]
```diff
@@ -1,3 +1,56 @@
...
+`int` / `tinyint` / `smallint` / `bigint` also have unsigned version, and there corresponding values ranges are:
+
+- `int unsigned`: 0~4294967295
+- `tinyint unsigned`: 0~255
+- `smallint unsigned`: 0~65535
+- `bigint unsigned`: 0~18446744073709551615
+
+## Variable-sized type limitations
+
+The max capacities of variable-sized type, like `string` and `varbinary` are determined by their encodings and how storage engine handles them.
```
Comment (xtang) [inline docs/reference/data-types.md:29]:
```suggestion
The max capacities of variable-sized type, such as `string` and `varbinary`, are determined by their encodings and how storage engine handles them. 
```

File: docs/reference/data-types.md [hunk 2]
```diff
@@ -1,3 +1,56 @@
 # Data Types
 
-TODO
+GreptimeDB supports following data types:
```
Comment (Wenjie0329) [inline docs/reference/data-types.md:3]:
```suggestion
GreptimeDB supports following data types:
```

File: docs/reference/data-types.md [hunk 3]
```diff
@@ -1,3 +1,57 @@
...
+|`int`| 32-bit signed integers between -2147483648~2147483647| `integer`|
+|`bigint`| 64-bit signed big integers between -9223372036854775808~9223372036854775807| |
+|`varchar`|UTF-8 encoded strings|`text`/`string`/`char `|
+|`float`|32-bit IEEE754 floating point values ||
+|`double`|Double precision IEEE 754 floating point values||
+|`boolean`|Boolean values||
+|`varbinary`|Variable length binary values||
+|`date`|32-bit date values||
+|`datetime`|64-bit datetime values||
+|`timestamp`|Timestamp values with millisecond precision||
```
Comment (waynexia) [inline docs/reference/data-types.md:18]:
```suggestion
|`timestamp`| 64-bit timestamp values with millisecond precision||
```

**Review by xtang — APPROVED** [review]

**Comment by xtang** [comment]
LGTM. Please help to merge this pr when you feel good too. @waynexia

**Pull request merged**